optuna/_dataframe.py

```diff
@@ -0,0 +1,83 @@
+import collections
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Set
+from typing import Tuple
+
+import optuna
+from optuna._imports import try_import
+from optuna.trial._state import TrialState
+
+
+with try_import() as _imports:
+    # `trials_dataframe` is disabled if pandas is not available.
+    import pandas as pd  # NOQA
+
+# Required for type annotation in `Study.trials_dataframe`.
+if not _imports.is_successful():
+    pd = object  # type: ignore # NOQA
+
+
+def _trials_dataframe(
+    study: "optuna.Study",
+    attrs: Tuple[str, ...],
+    multi_index: bool,
+) -> "pd.DataFrame":
```

Comment:
Do we need the trailing comma? The number of arguments is not so large, and I'm not sure if it is needed here.
```suggestion
    study: "optuna.Study", attrs: Tuple[str, ...], multi_index: bool
) -> "pd.DataFrame":
```

Reply:
You're right, let's remove the comma.